### PR TITLE
feat: rename packages to Recommend

### PR DESCRIPTION
**Pull request by francoischalifour**
Long term, we want to have all Algolia Recommend-related APIs under these packages, so this PR renames all our modules from `@algolia/recommendations-*` to `@algolia/recommend-*`.

This decision was approved by @mthuret.

(Note that the JS `getRecommendations` and React `useRecommendations` APIs weren't renamed.)

### CODE REVIEW

**Review by Haroenv — APPROVED** [review]
looks like this is correct 👍

File: packages/recommend-react/README.md [hunk 1]
```diff
@@ -254,12 +254,9 @@ function App({ currentObjectID }) {
   return (
     <div className="auc-Recommendations">
```
Comment (shortcuts) [inline packages/recommend-react/README.md]:
```suggestion
    <div className="auc-Recommend">
```

(there's other occurrences of this class)

**Review by shortcuts — APPROVED** [review]
https://github.com/algolia/ui-components/blob/ea7c08aa99fcb9ff6eb8bae009f315ccabedd8f5/packages/recommend-vdom/src/DefaultChildren.tsx#L13

I think this class should also be `auc-Recommend`, otherwise everything seems good!

File: packages/recommend-react/README.md (hunk 1 above)
Comment (francoischalifour) [reply packages/recommend-react/README.md]: Good catch—fixed in 3c562450fc18f8f6628385764b4d6f1dddb0b0ad.

**Review by shortcuts — APPROVED** [review]

**Pull request merged**